llvm/test/CodeGen/AMDGPU/fneg-combines.f16.ll

```diff
@@ -530,7 +530,7 @@ define amdgpu_ps half @fneg_fadd_0_f16(half inreg %tmp2, half inreg %tmp6, <4 x
 ; VI-SAFE-NEXT:    v_rcp_f16_e32 v0, s1
 ; VI-SAFE-NEXT:    v_mov_b32_e32 v1, s0
 ; VI-SAFE-NEXT:    v_mul_f16_e32 v0, 0, v0
-; VI-SAFE-NEXT:    v_add_f16_e32 v0, 0, v0
+; VI-SAFE-NEXT:    v_fma_f16 v0, v0, 0, 0
```

Comment:
Based on the test names, I think it would be better to just add contract flags to these instead of using the global flag